webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/Compliance.scala

```diff
@@ -112,9 +115,84 @@ final case class ByRuleValueCompliance(
 final case class ByRuleNodeCompliance(
     id    : NodeId
   , name  : String
+  , compliance: ComplianceLevel
   , values: Seq[ComponentValueStatusReport]
 )
 
+final case class ByRuleByNodeCompliance(
+  id        : NodeId
+  , name      : String
+  , compliance: ComplianceLevel
+  , directives: Seq[ByRuleByNodeByDirectiveCompliance]
+)
+
+final case class ByRuleByNodeByDirectiveCompliance(
+  id        : DirectiveId
+  , name      : String
+  , compliance: ComplianceLevel
+  , components: Seq[ByRuleByNodeByDirectiveByComponentCompliance]
+)
+
+sealed trait ByRuleByNodeByDirectiveByComponentCompliance {
+  def name : String
+  def compliance : ComplianceLevel
+}
+
+final case class ByRuleByNodeByDirectiveByBlockCompliance(
+  name      : String
+  , compliance: ComplianceLevel
+  , subComponents     : Seq[ByRuleByNodeByDirectiveByComponentCompliance]
+) extends ByRuleByNodeByDirectiveByComponentCompliance
+
+final case class ByRuleByNodeByDirectiveByValueCompliance(
+  name      : String
+  , compliance: ComplianceLevel
+  , values     : Seq[ComponentValueStatusReport]
+) extends ByRuleByNodeByDirectiveByComponentCompliance
+
+object ByRuleByNodeCompliance {
+  def fromDirective(directives : Seq[ByRuleDirectiveCompliance]) = {
+    final case class TmpStruct (directiveId: DirectiveId, blocks: List[String], component:  String, node : ByRuleNodeCompliance)
+    def recurseComponent( byRuleComponentCompliance: ByRuleComponentCompliance): Seq[(NodeId,ByRuleByNodeByDirectiveByComponentCompliance)] = {
```

Comment:
a bit of comments here would be much appreciated when we'll have to delve into it in the future

Reply:
Yes I will add them after I tried this in real life, it compiles, and the logic in my head is ok, but I need to confront it in real life (the PR is still a wip)

Reply:
is it time, now :)